test/e2e/api_ledgers_list_test.go

```diff
@@ -26,26 +27,80 @@ var _ = Context("Ledger engine tests", func() {
 				Debug:                 debug,
 			},
 			NatsURL:         natsServer.GetValue().ClientURL(),
-			MaxPageSize:     5,
-			DefaultPageSize: 5,
+			MaxPageSize:     15,
+			DefaultPageSize: 15,
 		}
 	})
 
 	When("creating 10 ledger", func() {
 		BeforeEach(func() {
-			for i := range 10 {
+			for i := range 20 {
 				err := CreateLedger(ctx, testServer.GetValue(), operations.V2CreateLedgerRequest{
 					Ledger: fmt.Sprintf("ledger%d", i),
+					V2CreateLedgerRequest: components.V2CreateLedgerRequest{
+						Bucket: pointer.For(fmt.Sprintf("bucket%d", i%2)),
+						Metadata: map[string]string{
+							"foo": fmt.Sprintf("bar%d", i%3),
+						},
+					},
 				})
 				Expect(err).To(BeNil())
 			}
 		})
-		It("should be listable", func() {
+		It("should be listable without filter", func() {
 			ledgers, err := ListLedgers(ctx, testServer.GetValue(), operations.V2ListLedgersRequest{
 				PageSize: pointer.For(int64(100)),
 			})
 			Expect(err).To(BeNil())
-			Expect(ledgers.Data).To(HaveLen(5))
+			Expect(ledgers.Data).To(HaveLen(15))
+		})
+		It("filtering on bucket should return 5 ledgers", func() {
+			ledgers, err := ListLedgers(ctx, testServer.GetValue(), operations.V2ListLedgersRequest{
+				PageSize: pointer.For(int64(100)),
+				RequestBody: map[string]any{
+					"$match": map[string]any{
+						"bucket": "bucket0",
+					},
+				},
+			})
+			Expect(err).To(BeNil())
+			Expect(ledgers.Data).To(HaveLen(10))
+		})
+		It("filtering on metadata[foo] = 0 should return 7 ledgers", func() {
+			ledgers, err := ListLedgers(ctx, testServer.GetValue(), operations.V2ListLedgersRequest{
+				PageSize: pointer.For(int64(100)),
+				RequestBody: map[string]any{
+					"$match": map[string]any{
+						"metadata[foo]": "bar0",
+					},
+				},
+			})
+			Expect(err).To(BeNil())
+			Expect(ledgers.Data).To(HaveLen(7))
+		})
+		It("filtering on name = ledger0 should return 1 ledger", func() {
+			ledgers, err := ListLedgers(ctx, testServer.GetValue(), operations.V2ListLedgersRequest{
+				PageSize: pointer.For(int64(100)),
+				RequestBody: map[string]any{
+					"$match": map[string]any{
+						"name": "ledger0",
+					},
+				},
+			})
+			Expect(err).To(BeNil())
+			Expect(ledgers.Data).To(HaveLen(1))
+		})
+		It("filtering on name = ledger0 should return 1 ledger", func() {
```

Comment:
Test name is a duplicate of the above, but the test content is different 👀

Reply:
Good catch! It was a copy/paste.